src/core/geocms/qgsgeocmsconnection.cpp

```diff
@@ -0,0 +1,98 @@
+/***************************************************************************
+    qgsgeocmsconnection.cpp
+    ---------------------
+    begin                : Feb 2017
+    copyright            : (C) 2017 by Muhammad Yarjuna Rohmat, Ismail Sunni
+    email                : rohmat at kartoza dot com, ismail at kartoza dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsgeocmsconnection.h"
+#include "qgssettings.h"
+#include "qgslogger.h"
+#include "qgsnetworkaccessmanager.h"
+
+#include <QMultiMap>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+#include <QByteArray>
+#include <QJsonDocument>
+#include <QDebug>
+#include <QUrl>
+#include <QDomDocument>
+
+
+QgsGeoCmsConnection::QgsGeoCmsConnection( const QString &geoCMSName, const QString &connName )
+  : mGeoCMSName( geoCMSName )
+  , mConnName( connName )
+{
+  QgsDebugMsg( "theConnName = " + connName );
+
+  QgsSettings settings;
+
+  QString key = "qgis/connections-" + mGeoCMSName.toLower() + '/' + mConnName;
```

Comment:
You might want to use the namespaces available in QgsSettings, have a look to https://qgis.org/api/classQgsSettings.html#aba2f174ee000905cac8aff26665ed7c7
providers, seems appropriate here.